examples/tests/valid/json-types.test.w

```diff
@@ -0,0 +1,25 @@
+
+
+let extra = MutArray<Json>["bar"];
+let base = [Json "foo"];
+base.concat(extra.copy());
+
+let base2 = Array<Json>["foo"];
+
+let obj = {
+  first: "ok",
+  second: ["cool"].concat(base2)
+};
+
+let arr = MutArray<num>[1, 2, 3];
+arr.push(4);
+
+let map = MutMap<str>{"a" => "1"};
+map.set("b", "2");
+
+let takeJson = (obj: Json) => {
+  
+};
+
+takeJson({ values: arr });
+takeJson({ values: map });
```

Comment:
These `takeJson` calls shouldn't be valid. MutArray isn't safe to convert directly to Json (for the same reason `MutArray` can't passed to a function that expects an `Array`). But it would be ok to implicitly convert a `MutArray` to `MutJson`.

Reply:
Fixed-
could not get to the implicit conversion since `MutJson` and `Json` behave strangely- #7002 #7003